### ISSUE TITLE
false negative on tests

### ISSUE THREAD

**Issue by 91khr** [rust-lang/rust-clippy#8861]
### Summary

Code in `#[cfg(test)]` won't be linted unless with `--fix --allow-dirty`.

### Lint Name

Only tested for dead_code and unused_io_amount, but seems that all lints have this bug

### Reproducer

1. `cargo new t`
1. `main.rs`:
    ```rust
    fn main() {
        println!("Hello, world!");
    }
    
    #[cfg(test)]
    mod tests {
        struct Unused;
    }
    ```
1. `cargo clippy` produced no warnings, but `cargo clippy --fix --allow-dirty` detected the unused struct.

### Version

`cargo clippy --version`: clippy 0.1.62 (c52b9c1 2022-05-16)
`rustc -Vv`:
```text
rustc 1.63.0-nightly (c52b9c10b 2022-05-16)
binary: rustc
commit-hash: c52b9c10bfb5164015eb977ff498e0597ae63eb1
commit-date: 2022-05-16
host: x86_64-unknown-linux-gnu
release: 1.63.0-nightly
LLVM version: 14.0.4
```

**Comment by Alexendoo** [comment]
`cargo clippy` is like `cargo check`, which doesn't check tests by default, you can pass `--tests` or `--all-targets` for that

`cargo clippy --fix` is like `cargo fix`, which has `--all-targets` applied by default

**Comment by giraffate** [comment]
Thanks for the report!

I'm closing this, because https://github.com/rust-lang/rust-clippy/issues/8861#issuecomment-1133602195 seems to be the answer. If you have any question, feel free to comment here or file an another issue.
